packages/runtime-common/realm-auth-client.ts

```diff
@@ -141,31 +141,55 @@ export class RealmAuthClient {
     if (!userId) {
       throw new Error('userId is undefined');
     }
-    return this.fetch(`${this.realmURL.href}${this.sessionEndpoint}`, {
-      method: 'POST',
-      headers: {
-        Accept: 'application/json',
-      },
-      body: JSON.stringify({
-        user: userId,
+    return this.withRetries(() =>
+      this.fetch(`${this.realmURL.href}${this.sessionEndpoint}`, {
+        method: 'POST',
+        headers: {
+          Accept: 'application/json',
+        },
+        body: JSON.stringify({
+          user: userId,
+        }),
       }),
-    });
+    );
   }
 
   private async challengeRequest(
     challenge: string,
     challengeResponse?: string,
   ) {
-    return this.fetch(`${this.realmURL.href}${this.sessionEndpoint}`, {
-      method: 'POST',
-      headers: {
-        Accept: 'application/json',
-      },
-      body: JSON.stringify({
-        user: this.matrixClient.getUserId(),
-        challenge,
-        ...(challengeResponse ? { challengeResponse } : {}),
+    return this.withRetries(() =>
```

Comment:
Can this come in handy? https://github.com/cardstack/boxel/blob/38dd36fe2512c41e143d9ce8a6b2e45e3793f5f8/packages/runtime-common/utils.ts#L3-L24

Reply:
almost--this mechanism doesn't have a back off. maybe we could add a backoff function to this?